### PR TITLE
Release v0.1.0

### PR DESCRIPTION
**Pull request by jason-jackson**
Add basic auth proxy server

### CODE REVIEW

File: codeship-services.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    - ./Caddyfile:/etc/caddy/Caddyfile
+  command: caddy run
+
+test:
+  build:
+    context: .
+    dockerfile: Dockerfile-test
+  environment:
+    AUTH_COOKIE_NAME: _proxy
+    AUTH_TOKEN_SECRET: Rm9yIEdvZCBzbyBsb3ZlZCB0aGUgd29ybGQgdGhhdCBoZSBnYXZlIGhpcyBvbmUgYW5kIG9ubHkgU29uLCB0aGF0IHdob2V2ZXIgYmVsaWV2ZXMgaW4gaGltIHNoYWxsIG5vdCBwZXJpc2ggYnV0IGhhdmUgZXRlcm5hbCBsaWZlLiAtIEpvaG4gMzoxNg==
```
Comment (Baggerone) [inline codeship-services.yml:25]: Sorry if I've already asked .... This isn't being used anywhere in the real world, is it?

**Review by Baggerone — APPROVED** [review]
Better wait for Steve S. to review it too.

File: Caddyfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	order dynamic_proxy first
+}
+
+:8888 {
+	encode zstd gzip
+
+	respond /test 200 {
+		body {vars.upstream}
+		close
+	}
```
Comment (forevermatt) [inline Caddyfile:11]: Reminder: Remove this

File: codeship-services.yml (hunk 1 above)
Comment (jason-jackson) [reply codeship-services.yml:25]: Not that I'm aware of, I could change it to something more obviously fake though.

**Comment by jason-jackson** [comment]
I'm also working on another PR that I'd like to include in here, so feel free to hold off if you wish.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
+version: "3.7"
+
+services:
+  app:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "8888:8888"
```
Comment (Schparky) [inline docker-compose.yml:9]: 8888 is somewhat commonly used. I have an list of "reserved" ports for ITSE dev work. I just allocated 53060 for this project. You can use it if you're interested in not having potential port conflicts.

See https://docs.google.com/spreadsheets/d/1g0uS4xPfrEGUyYu4c9iaOK8XXYIW8Ymi_a0_OmkpOf4/edit#gid=0

**Review by Schparky — APPROVED** [review]

**Review by Schparky — APPROVED** [review]

**Pull request merged**